### PR TITLE
Handle proximity radius in /autocomplete endpoint

### PR DESCRIPTION
**Pull request by AzHicham**

### CODE REVIEW

File: libs/mimir/src/adapters/primary/common/filters.rs [hunk 1]
```diff
@@ -14,4 +14,7 @@ pub struct Filters {
     pub poi_types: Option<Vec<String>>,
     pub limit: i64,
     pub timeout: Option<Duration>,
+    pub proximity_scale: Option<f64>,
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/common/filters.rs:17]: I think you should have a struct proximity with three parameters to indicate they work together

Comment (crocme10) [reply libs/mimir/src/adapters/primary/common/filters.rs:17]: (pretty much like the struct decay)

File: libs/mimir/src/adapters/primary/common/dsl.rs [hunk 2]
```diff
@@ -85,14 +77,24 @@ fn build_filters(
...
     boosts.push(admin_weight_boost);
-    let proximity_boost =
-        coord.map(|coord| build_proximity_boost(coord, &settings.importance_query.proximity.decay));
+    let setting_decay = &settings.importance_query.proximity.decay;
+    let proximity_boost = filters.coord.clone().map(|coord| {
+        build_proximity_boost(
+            coord,
+            &Decay {
+                scale: filters.proximity_scale.unwrap_or(setting_decay.scale),
+                offset: filters.proximity_scale.unwrap_or(setting_decay.offset),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/common/dsl.rs:92]: probably meant proximity_offset

File: libs/mimir/src/adapters/primary/common/dsl.rs [hunk 3]
```diff
@@ -85,14 +77,24 @@ fn build_filters(
...
+                decay: filters.proximity_scale.unwrap_or(setting_decay.decay),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/common/dsl.rs:93]: ditto

File: libs/mimir/src/adapters/primary/bragi/api.rs [hunk 4]
```diff
@@ -45,6 +45,9 @@ pub struct ForwardGeocoderQuery {
     pub pt_dataset: Option<Vec<String>>,
     pub poi_dataset: Option<Vec<String>>,
     pub request_id: Option<String>,
+    pub proximity_scale: Option<f64>,
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/api.rs:48]: probably need to extract this in separate struct, and just add a `pub proximity: Option<Proximity>`

File: libs/mimir/src/adapters/primary/common/filters.rs (hunk 1 above)
Comment (AzHicham) [reply libs/mimir/src/adapters/primary/common/filters.rs:17]: Unfortunately using  serde_flatten didn't work well ...

File: libs/mimir/src/adapters/primary/common/dsl.rs (hunk 2 above)
Comment (AzHicham) [reply libs/mimir/src/adapters/primary/common/dsl.rs]: fixed

File: libs/mimir/src/adapters/primary/common/filters.rs (hunk 1 above)
Comment (AzHicham) [reply libs/mimir/src/adapters/primary/common/filters.rs]: Finally I managed to make it work

File: libs/mimir/src/adapters/primary/bragi/api.rs (hunk 4 above)
Comment (AzHicham) [reply libs/mimir/src/adapters/primary/bragi/api.rs]: Fixed thanks !!

**Review by azime — APPROVED** [review]

**Pull request merged**